contracts/test/behaviour/strategy.js

```diff
@@ -0,0 +1,401 @@
+const { expect } = require("chai");
+const { Wallet } = require("ethers");
+
+const { units, usdcUnits } = require("../helpers");
+const { impersonateAndFund } = require("../../utils/signers");
+const { parseUnits } = require("ethers/lib/utils");
+
+/**
+ *
+ * @param {*} context a function that returns a fixture with the additional properties:
+ * - strategy: the strategy to test
+ * - assets: array of tokens to test
+ * - valueAssets: optional array of assets that work for checkBalance and withdraw. defaults to assets
+ * - vault: Vault or OETHVault contract
+ * - harvester: Harvester or OETHHarvester contract
+ * @example
+    shouldBehaveLikeStrategy(() => ({
+      ...fixture,
+      strategy: fixture.fraxEthStrategy,
+      assets: [fixture.frxETH, fixture.weth],
+      valueAssets: [fixture.frxETH],
+      harvester: fixture.oethHarvester,
+      vault: fixture.oethVault,
+    }));
+ */
+const shouldBehaveLikeStrategy = (context) => {
```

Comment:
We should also check that only the harvester can call `collectRewardTokens` on the strategy.

Reply:
that is indirectly covered in the harvester behaviour tests. Not all strategies have rewards to harvest so I'd prefer to keep it out of the strategy behaviour.
I'll add explicit checks that calls to `collectRewardTokens` succeed from the harvester and fail if not to the harvester behaviour tests.